airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresUtils.java

```diff
@@ -31,18 +31,21 @@ public static boolean isCdc(final JsonNode config) {
   }
 
   public static Duration getFirstRecordWaitTime(final JsonNode config) {
-    if (config.has("initial_waiting_seconds")) {
-      final int seconds = config.get("initial_waiting_seconds").asInt();
+    final JsonNode replicationMethod = config.get("replication_method");
+    Duration firstRecordWaitTime = DEFAULT_FIRST_RECORD_WAIT_TIME;
+    if (replicationMethod != null && replicationMethod.has("initial_waiting_seconds")) {
+      final int seconds = config.get("replication_method").get("initial_waiting_seconds").asInt();
       if (seconds > MAX_FIRST_RECORD_WAIT_TIME.getSeconds()) {
         LOGGER.warn("First record waiting time is overridden to {} minutes, which is the max time allowed for safety.",
```

Comment:
completely unrelated to this change so feel free to ignore - could this happen in checkConnection rather than at sync time?

Reply:
Good point. I will add it to the connection check.

Reply:
Done.